runtime/jvmti/jvmtiHelpers.c

```diff
@@ -147,6 +147,10 @@ getVMThread(J9VMThread *currentThread, jthread thread, J9VMThread **vmThreadPtr,
 		isThreadAlive = J9VMJAVALANGTHREAD_STARTED(currentThread, threadObject) && (NULL != targetThread);
 	}
 
+#if JAVA_SPEC_VERSION >= 19
+	Assert_JVMTI_true((NULL != targetThread) || isVirtualThread);
```

Comment:
Misjudgment on my part. We will need to relocate the assert after `omrthread_monitor_exit(vm->vmThreadListMutex)` (line 178). 

For `rc == JVMTI_ERROR_THREAD_NOT_ALIVE` i.e. `!isThreadAlive`, `targetThread` will be NULL.